bin/brew

```diff
@@ -21,7 +24,32 @@ then
                          chdir "$BREW_SYMLINK_DIRECTORY" && pwd -P)"
 fi
 
-BREW_LIBRARY_DIRECTORY="$(chdir "$BREW_FILE_DIRECTORY"/../Library && pwd -P)"
+HOMEBREW_PREFIX="$(chdir "$(dirname "$(dirname "$HOMEBREW_BREW_FILE")")" && pwd -P)"
+HOMEBREW_REPOSITORY="$(chdir "$BREW_FILE_DIRECTORY"/../ && pwd -P)"
+HOMEBREW_LIBRARY="$(chdir "$BREW_FILE_DIRECTORY"/../Library && pwd -P)"
+
+# Where we store built products; /usr/local/Cellar if it exists,
+# otherwise a Cellar relative to the Repository.
+if [ -d "$HOMEBREW_PREFIX/Cellar" ]
+then
+  HOMEBREW_CELLAR="$HOMEBREW_PREFIX/Cellar"
+else
+  HOMEBREW_CELLAR="$HOMEBREW_REPOSITORY/Cellar"
+fi
+
+if [ "$*" = "--prefix" ]
+then
+  echo "$HOMEBREW_PREFIX"
+  exit 0
+elif [ "$*" = "--cellar" ]
+then
+  echo "$HOMEBREW_CELLAR"
+  exit 0
+elif [ "$*" = "--repository" ] || [ "$*" = "--repo" ]
```

Comment:
@xu-cheng How can we make `brew command --repository` working if it’s not a file anymore? What should we output?